experimental/framework/fluid-llm/src/shared-tree-diff/sharedTreeDiff.ts

```diff
@@ -0,0 +1,424 @@
+import { type TreeArrayNode, NodeKind } from "@fluidframework/tree";
+
+import { isTreeMapNode } from "./utils.js";
+
+/**
+ * Represents a path through a tree of objects.
+ * number values represent array indices whereas string values represent object keys.
+ */
+export type ObjectPath = (string | number)[];
+
+/**
+ * Represents a create operation between two branches of a tree.
+ * Meaning that an attribute (a shared tree node) was identified as being created.
+ */
+export interface DifferenceCreate {
```

Comment:
You've effectively created a schema for describing tree edits here. Have you explored having the AI agent generate these directly, rather than generating the new data state? That would allow us to remove the diff deriving logic below.